src/opm/io/eclipse/ESmry.cpp

```diff
@@ -304,10 +307,21 @@ ESmry::ESmry(const std::string &filename, bool loadBaseRunData)
             
             i++;
 
-            EclFile resfile(std::get<1>(arraySourceList[i]));
+            if (std::get<1>(arraySourceList[i]) != prevFile){
+                
+                if (pEclFile != nullptr){
```

Comment:
you don't need this, it happens on itself if you assign a new value to the unique_ptr.